### ISSUE TITLE
Makefile for submodules contains local paths

### ISSUE THREAD

**Issue by GuillaumeHolley** [mobinasri/flagger#3]
Hi,

I noticed that the [Makefile](https://github.com/mobinasri/flagger/blob/main/programs/Makefile) for submodules contains such as `/usr/local/include/htslib` or `/home/apps/sonLib/lib/sonLib.a`. I imagine that htslib and sonLib should be integrated as submodules too.

Guillaume

**Comment by GuillaumeHolley** [comment]
I also get errors while trying to compile after fixing the `htslib` and `sonLib` library/include paths manually.

First, I had to specify `-std=c99` in the Makefile, otherwise some for loop constructions would not compile. Then a few files in `src/` require to add `#include <sys/types.h>` in the header as the type `ssize_t` is unknown to them otherwise. Now, `src/filter_alt_reads.c` cannot find `cigar_it.h` because it is in the `submodule` directory and not the `src` one.

**Comment by GuillaumeHolley** [comment]
`cigar_it` is missing as a submodule but is included in the [submodules of Secphase](https://github.com/mobinasri/secphase/tree/main/programs/submodules).

**Comment by mobinasri** [comment]
Hi @GuillaumeHolley 
I recommend using the docker image "mobinasri/flagger:v0.1" (or "mobinasri/flagger:dev-v0.1" but this one is not stable) for running Flagger and using the internal programs. 
Beside that using WDL files is even more convenient if you are familiar how to run WDL files through [Cromwell](https://cromwell.readthedocs.io/en/stable/CommandLine/).
The submodules and libraries essential for using the Makefile are being imported and installed in [this Dockerfile](https://github.com/mobinasri/flagger/blob/v0.1/Dockerfile). If you want to build the project from scratch it's better to follow whatever is being done in the Dockerfile. For example the `cigar_it` library is imported from secphase in [this line](https://github.com/mobinasri/flagger/blob/ab82fa88d58a7916ec1bea7193c2728c6bcc371b/Dockerfile#L48)

**Comment by GuillaumeHolley** [comment]
Hi @mobinasri,

Thanks for the feedback!

For reasons that are particular to my compute cluster, I am unable to run WDL files at the moment so I had to improvise. I appreciate the advice though and I'll make sure to use the WDL pipeline through Cromwell if/when it is possible. When it comes to installing the programs, I noticed a bit late indeed that those were already in the Docker image so I bothered installing them from scratch for nothing. I think what confused me a little bit was that they are 2 READMEs, [here](https://github.com/mobinasri/flagger) and [here](https://github.com/mobinasri/flagger/tree/main/docs/flagger), and even though the first README refers to the WDL pipeline for running Flagger, both also contains precise descriptions and command lines so it seemed to me I could "bypass" the WDL pipeline and just follow the step by step descriptions. However, the descriptions are an overview of the pipeline and and naturally, some steps from the WDL pipeline are not in there. So maybe just as a suggestion, I think this [README ](https://github.com/mobinasri/flagger) should make it clear that the programs are not intended to be installed directly but instead, used through the Docker image. Also, maybe mentioning that the described pipeline in this [README](https://github.com/mobinasri/flagger/tree/main/docs/flagger) is an overview with some smaller steps skipped and one must run the WDL pipeline instead. Or maybe I misunderstood the whole thing completely :)

PS: This README mentions `quay.io/masri2019/hpp_coverage:latest` as the Docker image, should it be `mobinasri/flagger:v0.1` instead?
